Remove gulp, add postcss

occupant opened the pull request:
Removes
- gulpfile.js (deprecated)
- src/sass/_mixins.scss (deprecated)
- src/sass/_functions.scss (deprecated)

Adds
- .stylelintrc.json (stylelint config)
- postcss.config.js (postcss config)
- package.scripts.js (placeholder for node scripts)
- tailwind.config.js (tailwind config)

Renames
- all .scss files to .css

occupant committed:
Remove gulp, add postcss

Removes
- gulpfile.js (deprecated)
- src/sass/_mixins.scss (deprecated)
- src/sass/_functions.scss (deprecated)

Adds
- .stylelintrc.json (stylelint config)
- postcss.config.js (postcss config)
- package.scripts.js (placeholder for node scripts)
- tailwind.config.js (tailwind config)

Renames
- all .scss files to .css

occupant commented:
Hi both, this is in reference to Issue #45 and #44. I'll add notes later this afternoon to assist with the review.

richardtape commented on src/css/abstracts/variables.css:
```diff
@@ -0,0 +1,72 @@
+* @type List
+*/
+$text-font-stack: 'Open Sans', 'Helvetica Neue Light', 'Helvetica Neue', 'Helvetica', 'Arial', sans-serif;
+
```
Do we have a need for a serif stack?

occupant commented:
This is the minimum set of changes I thought could be made in switching from gulp to node.

Assumptions:

- #44 we have agreement on moving forward with stylelint
- #45 we have agreement on moving forward with node.js scripts in place of gulp
- #45 we have agreement on moving forward with postcss in place of SASS
- #45 we have agreement on moving forward with tailwindcss (implied and included, but not required)

The most important changes are all visible in the package.json. I have removed the following dependencies:

- gulp
- gulp-rename
- gulp-sass
- gulp-sourcemaps
- gulp-postcss

And added these as new dependencies:

- tailwindcss
- @fullhuman/postcss-purgecss
- cross-env
- cssnano
- postcss-cli
- postcss-custom-properties
- postcss-import
- postcss-simple-vars
- stylelint
- stylelint-config-rational-order
- stylelint-config-standard
- stylelint-order

Also added are the new scripts that basically replace gulp . Descriptions for these are in the readme file.

- `lint:css`
- `lint:fix-css`
- `build:css`
- `build:js`
- `build:dev`
- `developmen`
- `build:prod`
- `production`

richardtape commented:
Thumbs up from me!

hazuremon commented on .stylelintrc.json:
```diff
@@ -0,0 +1,45 @@
+		"declaration-empty-line-before": "never",
+		"function-linear-gradient-no-nonstandard-direction": null,
+		"function-whitespace-after": null,
+		"indentation": "tab",
```
We currently agreed on using spaces of 4 as indentation, but i agreed on switching to tab.

Already submitted an issue to the [Guide repo](https://github.com/UBCCM/guides) on why tab maybe a better option.

occupant replied:
Our [editorconfig](https://github.com/UBCCM/UBC-DEF/blob/6e695ed00bdbf6a973f41fd0e262db17c065497b/.editorconfig#L10) has tabs as the default already (with the tab sized to 4 spaces), so I think we're good.

occupant commented:
We do (and it exists in some form in the tailwind config) but I didn’t want to change anything, just convert. Happy to add though if everyone is in agreement

—

James O’Toole
Programmer Analyst, Web Services
Information Technology | Engage. Envision. Enable.
The University of British Columbia
604.827.2001 | james.otoole@ubc.ca

On Nov 19, 2019, at 11:05 AM, Richard Tape <notifications@github.com> wrote:

﻿

@richardtape commented on this pull request.

________________________________

In src/css/abstracts/variables.css<https://github.com/UBCCM/UBC-DEF/pull/47#discussion_r346579580>:

> @@ -0,0 +1,72 @@
+/**
+* -----------------------------------------------------------------------------
+* This file contains all application-wide variables.
+* -----------------------------------------------------------------------------
+*/
+
+/**
+* Regular font family
+* @type List
+*/
+$text-font-stack: 'Open Sans', 'Helvetica Neue Light', 'Helvetica Neue', 'Helvetica', 'Arial', sans-serif;
+


Do we have a need for a serif stack?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/UBCCM/UBC-DEF/pull/47?email_source=notifications&email_token=AAA7YTMORD2J7YI7JJBMOGDQUQ2HXA5CNFSM4JNR4Y5KYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCLU2VQI#pullrequestreview-317303489>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAA7YTOIQ3R3WOY5GMRST5DQUQ2HXANCNFSM4JNR4Y5A>.

hazuremon commented on package.json:
```diff
@@ -1,35 +1,53 @@
+		"lint:css": "stylelint \"src/**/*.css\" || exit 0",
+		"lint:fix-css": "stylelint \"src/**/*.css\" --fix || exit 0",
+		"build:css": "cross-env NODE_ENV=development postcss src/css/MAIN.css -o build/css/styles.all.css",
+		"build:js": "babel src/html/js --out-file build/js/ubc.all.js --ignore \"src/html/js/libs/*\"",
```
Should we keep this out from this PR?

If we decide to keep it, should we include babel in dependencies?

occupant replied:
Yes, you're right, this shouldn't be there. It's a discussion for later. I'll fix this.

richardtape commented:
No, you’re right, converting everything over is the first step. We can talk about changes separately.

> On Nov 19, 2019, at 11:40 AM, James <notifications@github.com> wrote:
> 
> We do (and it exists in some form in the tailwind config) but I didn’t want to change anything, just convert. Happy to add though if everyone is in agreement 
> 
> — 
> 
> James O’Toole 
> Programmer Analyst, Web Services 
> Information Technology | Engage. Envision. Enable. 
> The University of British Columbia 
> 604.827.2001 | james.otoole@ubc.ca 
> 
> On Nov 19, 2019, at 11:05 AM, Richard Tape <notifications@github.com> wrote: 
> 
> ﻿ 
> 
> @richardtape commented on this pull request. 
> 
> ________________________________ 
> 
> In src/css/abstracts/variables.css<https://github.com/UBCCM/UBC-DEF/pull/47#discussion_r346579580>: 
> 
> > @@ -0,0 +1,72 @@ 
> +/** 
> +* ----------------------------------------------------------------------------- 
> +* This file contains all application-wide variables. 
> +* ----------------------------------------------------------------------------- 
> +*/ 
> + 
> +/** 
> +* Regular font family 
> +* @type List 
> +*/ 
> +$text-font-stack: 'Open Sans', 'Helvetica Neue Light', 'Helvetica Neue', 'Helvetica', 'Arial', sans-serif; 
> + 
> 
> 
> Do we have a need for a serif stack? 
> 
> — 
> You are receiving this because you authored the thread. 
> Reply to this email directly, view it on GitHub<https://github.com/UBCCM/UBC-DEF/pull/47?email_source=notifications&email_token=AAA7YTMORD2J7YI7JJBMOGDQUQ2HXA5CNFSM4JNR4Y5KYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCLU2VQI#pullrequestreview-317303489>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAA7YTOIQ3R3WOY5GMRST5DQUQ2HXANCNFSM4JNR4Y5A>. 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/UBCCM/UBC-DEF/pull/47?email_source=notifications&email_token=AAA4RUQD53EACEEYUSQ5RYTQUQ6JXA5CNFSM4JNR4Y5KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEEPPBYI#issuecomment-555675873>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAA4RURWJZINPYR6VV6PGETQUQ6JXANCNFSM4JNR4Y5A>.
>

hazuremon commented on tailwind.config.js:
```diff
@@ -0,0 +1,148 @@
+			'white': '#fff',
+			'ubc-blue': '#002145',
+			// rgb - (0,33,69)
+			'ubc-blue-sea': '#003A79',
```
May not be best to create new name for UBC colour.

Could we keep it consistent with how we differentiate them currently? Primary and Secondary colour, and just by number?
ubc-primary-1: #002145
ubc-secondary-1: #0055B7
ubc-secondary-2: #00A7E1

or ubc-brand-primary-1, ubc-brand-secondary-1, ubc-brand-secondary-2 if we want to be more explicit.

etc...

occupant replied:
I disagree. Working with the colours is challenging because the [names](https://brand.ubc.ca/guidelines/downloads/ubc-colour-and-fonts/) are too abstract. The difference between secondary-1, 2, 3, etc. is lost on me and really doesn't denote anything of value. 

That said, if we move to css variables for colours, more abstract is better, but I would hope I could make a reasonable guess as to how secondary-1 was different from secondary-2.

Not a hill I'm willing to die on, but I found in working with them it was MUCH easier to have a name that was reminiscent of the colour I could expect.

For now, why don't I remove the colours from the tailwind config and make this a separate discussion?

occupant committed:
Remove javascript compile tasks, update readme

occupant committed:
Update tailwind.config.js

Remove secondary colors for now until we have a consistent naming scheme.

hazuremon reviewed (approved):
Thanks James, looks great! Thanks for making all these changes.

occupant committed:
Update tailwind.config.js

Update, trivial commit to test signed commit

occupant commented:
Looks like Github Desktop [doesn't support signed commits](https://help.github.com/en/github/authenticating-to-github/signing-commits) (hence the block). I'm going to make another trivial commit to see if I can clear this up.

occupant committed:
Update README.md
